### PR TITLE
Swagger doc

### PR DESCRIPTION
**Pull request by elx-chad**

### CODE REVIEW

File: .swagger-codegen-ignore [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Swagger Codegen Ignore
```
Comment (petebuletza) [inline .swagger-codegen-ignore]: autogenerated file, I dont think this should be committed.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            }
+        }
+
+        stage('Configure Aritfactory') {
+            steps {
+                script {
+                    rtGradle.useWrapper = true
+
+                    if(env.BRANCH_NAME == 'master') {
+                        rtGradle.deployer repo: 'bins-release-local', server: server
```
Comment (petebuletza) [inline Jenkinsfile]: **libs-release-local**

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                        rtGradle.resolver repo: 'bins-release', server: server
+                    }
+                    else {
+                        rtGradle.deployer repo: 'bins-snapshot-local', server: server
```
Comment (petebuletza) [inline Jenkinsfile]: **libs-snapshot-local**

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                        rtGradle.resolver repo: 'bins-snapshot', server: server
```
Comment (petebuletza) [inline Jenkinsfile]: **libs-snapshot**

File: build.sbt [hunk 5]
```diff
@@ -0,0 +1,20 @@
+lazy val root = (project in file(".")).
```
Comment (petebuletza) [inline build.sbt]: autogenerated file?

File: gradle.properties [hunk 6]
```diff
@@ -0,0 +1,14 @@
+artifactMajorVersion = 0
+artifactMinorVersion = 1
+artifactPatchVersion = 0
+
+applicationGroupId = com.elyxor.cachethq
+applicationTitle = Cachet HQ Swagger API
```
Comment (petebuletza) [inline gradle.properties]: Since this is just the okhttp-gson version, this should be more like Cachet HQ okhttp-gosn client

**Comment by petebuletza** [comment]
There are also some issue with how it's publishing the pom and ivy information into Artifactory.  It has the version as `compile(group: 'com.elyxor', name: 'cachethq', version: 'cachethq-swagger-java-client')`

I cant actually tell how the jar name is being created, but it should look more like this:
`compile(group: 'com.elyxor', name: 'cachethq-api-okhttp-gson-client', version: '0.1.0-wip.swagger_doc')`

**Comment by petebuletza** [comment]
Also, PRs should not publish to Artifactory.  Branch/Master (and Release if we go with all three)

File: .swagger-codegen-ignore (hunk 1 above)
Comment (elx-chad) [reply .swagger-codegen-ignore]: Actually, I need this.  Bane of my existence for the past few hours.  Swagger-codegen will create its own gradlew, gradlew.bat, and drop the gradle wrapper back to gradle 2.6.   It'll muck up your environment.

Comment (petebuletza) [reply .swagger-codegen-ignore]: Right. I think thats why we need to do the generation into a separate folder, and not in the root folder.  Then that gradle project can be done as a subproject, or maybe just some shell commands from the parent gradle. We'll need to get creative.  And I had that same issue, which is why I was up til 2am on Monday.

**Pull request merged**